File: src/LeetCodeCS/MaximumSubarray.cs
public class MaximumSubarray
{
    public int MaxSubArray(int[] nums)
    {
        int result = Int32.MinValue;
        int sum = 0;

        for (int i = 0; i < nums.Length; i++)
        {
            sum += nums[i];
            result = Math.Max(result, sum);
            if (sum < 0)
            {
                sum = 0;
            }
        }
        return result;
    }
}

File: src/LeetCodePY/maximumSubArray.py
from typing import List

class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        if not nums:
            return 0

        result = nums[0]

        sum = 0

        for i in range(len(nums)):
            sum = max(sum + nums[i], nums[i])

            result = max(result, sum)

        return result